libpoolprotocols/PoolClient.h

```diff
@@ -12,10 +12,46 @@ namespace dev
 {
 	namespace eth
 	{
+		class PoolConnection
+		{
+		public:
+			typedef enum {
+				STRATUM = 0,
+				ETHPROXY,
+				ETHEREUMSTRATUM
+			} StratumProtocol;
```

Comment:
not sure if we better keep this in stratum only

Reply:
Then maybe a numeric field for protocol specific data.

unsigned m_protocol_specific; // or individual protocol use.

If we'll allow uris like stratum2:// or stratum1://, then different pool endpoints could use a different version of stratum, so I'd like to carry that info with the connection.